Detectors/ITSMFT/ITS/reconstruction/include/ITSReconstruction/CATracker.h

```diff
@@ -10,16 +10,18 @@
 #define ALIITSUCATRACKER_H
 
 #include <vector>
+#include <array>
 
 #include "ITSReconstruction/CAaux.h"
 #include "ITSReconstruction/CATrackingStation.h"
 #include "DetectorsBase/Track.h"
 
-using std::vector;
-
 namespace AliceO2 {
   namespace ITS {
     namespace CA {
+      using std::vector;
```

Comment:
It's "forbidden" in the sense that our [coding conventions](https://htmlpreview.github.io/?https://github.com/AliceO2Group/CodingGuidelines/blob/master/coding_guidelines.html#Using_declarations_and_directives
) do not distinguish between scoped and unscoped ones. I do not know the rationale for the rule as I was not there when it was written, however i do agree with it, since the scoping simply limits pollution to a given namespace, but does not remove the silent damage a `using std` statement in a header can do. For example if you have the following:

```
// In a.h
#include <string>

namespace A {
  using namespace std;
  void somethingUnrelated(string) {}
}
```

```
// in b.h
#include <cstdlib>
#include <cmath>

namespace A {
double foo(double a) {
  return abs(a);
}
}
```

```
// in main.cc
#include "a.h"
#include "b.h"
#include <iostream>

int main() {
  std::cout << A::foo(0.5) << std::endl;
  return 0;
}
```

your result will depend on the order you include a.h and b.h, which is obviously a "Bad Thing™".